core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1499,6 +1500,163 @@ takeEndBySeq patArr (Fold fstep finitial fextract) =
                     SplitOnSeqKRLoop s _ _ _ -> s
          in fextract st
 
+-- | Like 'takeEndBySeq' but discards the matched sequence.
+--
+-- /Pre-release/
+--
+{-# INLINE takeEndBySeq_ #-}
+takeEndBySeq_ :: forall m a b. (MonadIO m, Storable a, Enum a, Eq a) =>
+       Array.Array a
+    -> Fold m a b
+    -> Fold m a b
```

Comment:
Can be `Fold m a b -> Fold m a (b, Int)`?

Reply:
But I don't think backtracking of any kind should be baked in folds. It feels very gimiki.